agent/consul/state/catalog.go

```diff
@@ -3020,13 +3014,42 @@ func (s *Store) ServiceTopology(
 	if err != nil {
 		return 0, nil, fmt.Errorf("failed to query intentions for %s", sn.String())
 	}
+
+	var foundUpstreamNames = make(map[structs.ServiceName]bool)
```

Comment:
@freddygv Would this be preferred over returning an array of notFoundUpstreamNames in the [previous commit](https://github.com/hashicorp/consul/pull/10811/commits/2a37b868ccf394a1217a3d06a20b8017859555da#r685576288)?

A downside to doing it here is that I'm weaving more specific logic for upstreams which might make refactoring more difficult in the future